test/conformance/conformance_join_node.cpp

```diff
@@ -315,3 +316,106 @@ TEST_CASE("key_matching join_node invoke semantics") {
+};
+
+template <typename JoinPolicy, typename... Args>
+struct join_node_type_generator_impl<0, JoinPolicy, Args...> {
+    using type = oneapi::tbb::flow::join_node<std::tuple<Args...>, JoinPolicy>;
+};
+
+template <std::size_t N, typename JoinPolicy>
+using join_node_type_generator_t = typename join_node_type_generator_impl<N, JoinPolicy>::type;
+
+template <std::size_t N, typename JoinPolicy, std::size_t Index, typename... BodyToGenerate>
+struct make_join_node_helper {
+    template <typename... Bodies>
+    static join_node_type_generator_t<N, JoinPolicy> make(oneapi::tbb::flow::graph& g, BodyToGenerate... body, Bodies... bodies) {
+        return make_join_node_helper<N, JoinPolicy, Index - 1, BodyToGenerate...>::make(g, body..., bodies..., body...);
+    }
+};
+
+template <std::size_t N, typename JoinPolicy, typename... BodyToGenerate>
+struct make_join_node_helper<N, JoinPolicy, 0, BodyToGenerate...> {
+    template <typename... Bodies>
+    static join_node_type_generator_t<N, JoinPolicy> make(oneapi::tbb::flow::graph& g, BodyToGenerate..., Bodies... bodies) {
+        return join_node_type_generator_t<N, JoinPolicy>(g, bodies...);
+    }
+};
+
+template <std::size_t N, typename JoinPolicy, typename... BodyToGenerate>
+join_node_type_generator_t<N, JoinPolicy> make_join_node(oneapi::tbb::flow::graph& g, BodyToGenerate... body) {
+    return make_join_node_helper<N, JoinPolicy, N, BodyToGenerate...>::make(g, body...);
+}
+
+template <std::size_t N>
+struct tuple_assert_impl {
+    template <typename Tuple, typename Message>
+    static void make(const Tuple& tuple, const Message& message) {
+        CHECK_MESSAGE(std::get<N - 1>(tuple) == message, "Unexpected tuple element");
+        tuple_assert_impl<N - 1>::make(tuple, message);
+    }
+};
+
+template <>
+struct tuple_assert_impl<0> {
+    template <typename Tuple, typename Message>
+    static void make(const Tuple&, const Message&) {}
+};
+
+template <std::size_t N, typename Tuple, typename Message>
+void tuple_assert(const Tuple& tuple, const Message& message) {
+    tuple_assert_impl<N>::make(tuple, message);
+}
+
+template <typename JoinPolicy, std::size_t N, typename... Body>
+void test_join_node_with_n_inputs_impl(Body... body) {
+    static_assert(sizeof...(Body) <= 1, "Unexpected arguments");
+
+    using namespace oneapi::tbb::flow;
+    using join_node_type = join_node_type_generator_t<N, JoinPolicy>;
+    graph g;
+    int message = 42;
+
+    broadcast_node<int> submitter(g);
+
+    // A set of buffers between each input port for reserving policy
+    std::vector<buffer_node<int>> buffers(N, buffer_node<int>{g});
+
+    join_node_type join = make_join_node<N, JoinPolicy>(g, body...);
+    
+    using output_tuple = typename join_node_type::output_type;
+
+    std::size_t body_counter = 0;
+    function_node<output_tuple> function(g, serial, [&](const output_tuple& tuple) {
+        tuple_assert<N>(tuple, message);
```

Comment:
Rename to something more readable? For example,
```suggestion
        assert_all_items_equal_to<N>(tuple, message);
```
Similarly to `edge_maker`, this can also be extended to support other containers.
By the way, `std::get` already supports other collections: `std::array`, `std::pair`, `std::variant`, etc. So, such helper will automatically support them.
Therefore, consider renaming (since `tuple` could already be not necessarily a tuple) and moving this useful helper to some common place.